### PR TITLE
New: Add root folder select to EditSeriesModal

### LINKED ISSUE #5544: A way to change root path inside a show
### Is there an existing issue for this?

- [X] I have searched the existing open and closed issues

### Is your feature request related to a problem? Please describe

Every time i want to move a show i'm fixing, to another root path, or if i want to apply renaming to the folder name, i have to go back in main dashboard, search the show, use the multiselect to select it, then press the EDIT in the bottom and select the new or same root path, then apply

using the mass edit, but finding the show between thousands without using search (can't be used because it will open the show page, not filter the show list in dashboard), it's a pain

### Describe the solution you'd like

under the folder path inside a show edit window, i would like to have the same listbox as in the mass edit, to change the root path of that single show

### Describe alternatives you've considered

A button to rename the show folder, not the episodes. Will let rename the folder as per config, but no help in changing root folder if needed.

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by ricci2511**
#### Database Migration
NO

#### Description
Simply added the root folder select to the edit series modal. Changing the root folder also changes the series path accordingly.

#### Todos
- [ ] Tests
- [ ] Wiki Updates

#### Screenshot (if UI related)
root-folder1
root-folder2

#### Issues Fixed or Closed by this PR

* Closes #5544

### CODE REVIEW

File: frontend/src/Series/Edit/EditSeriesModalConnector.js [hunk 1]
```diff
@@ -2,14 +2,22 @@ import PropTypes from 'prop-types';
...
-  clearPendingChanges
+  clearPendingChanges,
+  fetchRootFolders
 };
 
 class EditSeriesModalConnector extends Component {
 
+  //
+  // Lifecycle
+  componentDidMount() {
```
Comment (markus101) [inline frontend/src/Series/Edit/EditSeriesModalConnector.js:17]: New line after comment

File: frontend/src/Series/Edit/EditSeriesModalContent.js [hunk 2]
```diff
@@ -148,6 +149,23 @@ class EditSeriesModalContent extends Component {
...
+              <FormLabel>{translate('Root Folder')}</FormLabel>
+
+              <FormInputGroup
+                type={inputTypes.ROOT_FOLDER_SELECT}
+                name="rootFolderPath"
+                {...rootFolderPath}
+                includeNoChange={true}
+                includeNoChangeDisabled={false}
+                selectedValueOptions={{ includeFreeSpace: false }}
+                helpText={translate(
```
Comment (markus101) [inline frontend/src/Series/Edit/EditSeriesModalContent.js:162]: This translation needs to be extracted to a key

File: frontend/src/Series/Edit/EditSeriesModalContentConnector.js [hunk 3]
```diff
@@ -80,7 +84,20 @@ class EditSeriesModalContentConnector extends Component {
...
+      isWindows,
+      title,
+      item: { path }
+    } = this.props;
+
+    dispatchSetSeriesValue({ name, value });
+
+    // Also update the path if the root folder path changes
+    if (name === 'rootFolderPath' && value !== 'noChange' && path.value.indexOf(value) === -1) {
+      const newPath = combinePath(isWindows, value, [title]);
```
Comment (markus101) [inline frontend/src/Series/Edit/EditSeriesModalContentConnector.js:98]: Using the series title isn't sufficient, this path needs to be generated on the backend to take the format into account.

**Review by markus101 — CHANGES_REQUESTED** [review]
The fact that you can change both the root folder and the path separately at the same time is going to cause some confusion and likely some issues, especially with how the new path is being updated.

Instead of a drop down a modal that either communicates with the backend and gets the new path given the chosen root folder and series or uses a pre-calculated value, like how it's done in `SeriesLookupController`, but the latter option could lead to a stale value.

**Comment by ricci2511** [comment]
Yeah, your concerns make sense. Just one thing, so if the root folder dropdown is ditched in favor of the modal approach how would the expected "user flow" be? Can you maybe provide a really high level example of the SeriesEditModal UI/UX? Just to make sure I understood correctly.

**Comment by markus101** [comment]
I think something like `Update Monitoring` when using series selection.

image

When the modal is opened the path for the series folder could be constructed, (precomputing could be useful for other things, but it's not a free operation and often not needed).

Once the root folder is chosen we could show the user the new path that will be used like we do in add series:
image

**Comment by markus101** [comment]
Closing, but we can reopen once updated.

**Pull request closed without merging**